sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -526,38 +528,44 @@ static az_result ready(az_event_policy* me, az_event event)
         az_mqtt5_property_bag_clear(&this_policy->_internal.property_bag);
         return AZ_ERROR_ARG;
       }
-      int32_t response_topic_length;
+      size_t response_topic_length = 0;
       _RETURN_AND_CLEAR_PROPERTY_BAG_IF_FAILED(
-          az_mqtt5_rpc_client_codec_get_response_topic(
+          az_mqtt5_rpc_client_codec_get_response_property_topic(
               this_policy->_internal.rpc_client_codec,
               event_data->rpc_server_client_id,
               _az_span_is_valid(event_data->command_name, 1, 0) ? event_data->command_name
                                                                 : AZ_SPAN_EMPTY,
-              this_policy->_internal.rpc_client_codec->_internal.response_topic_buffer,
+              (char*)az_span_ptr(this_policy->_internal.response_topic_buffer),
+              (size_t)az_span_size(this_policy->_internal.response_topic_buffer),
               &response_topic_length),
           &this_policy->_internal.property_bag);
+
       az_span response_topic_property_span = az_span_create(
-          az_span_ptr(this_policy->_internal.rpc_client_codec->_internal.response_topic_buffer),
-          response_topic_length - 1);
+          az_span_ptr(this_policy->_internal.response_topic_buffer),
+          (int32_t)response_topic_length - 1);
       _RETURN_AND_CLEAR_PROPERTY_BAG_IF_FAILED(
           az_mqtt5_property_bag_append_string(
               &this_policy->_internal.property_bag,
               AZ_MQTT5_PROPERTY_TYPE_RESPONSE_TOPIC,
               response_topic_property_span),
           &this_policy->_internal.property_bag);
 
+      size_t publish_topic_length = 0;
+
       _RETURN_AND_CLEAR_PROPERTY_BAG_IF_FAILED(
-          az_mqtt5_rpc_client_codec_get_request_topic(
+          az_mqtt5_rpc_client_codec_get_publish_topic(
               this_policy->_internal.rpc_client_codec,
               event_data->rpc_server_client_id,
               _az_span_is_valid(event_data->command_name, 1, 0) ? event_data->command_name
                                                                 : AZ_SPAN_EMPTY,
-              this_policy->_internal.rpc_client_codec->_internal.request_topic_buffer),
+              (char*)az_span_ptr(this_policy->_internal.request_topic_buffer),
+              (size_t)az_span_size(this_policy->_internal.request_topic_buffer),
```

Comment:
Does this cause any problems with the null terminator for mosquitto/paho?

Reply:
(since there's no slice)

Reply:
It is currently not causing any problems, but that's a good catch, I'll look into it!